src/modules/file/file.js

```diff
@@ -94,3 +94,277 @@ function file_entity_field_formatter_view(entity_type, entity, field, instance,
+    if (items[delta].children) {
+      items[delta].children.push({markup: html});
+    } else {
+      items[delta].children = [{markup: html}];
+    }
+
+    //drupalgap_add_js(drupalgap_get_path('module', 'file') + '/file_upload.js');
+  }
+
+  catch (error) {
+    console.log('file_entity_field_formatter_view - ' + error);
+  }
+}
+
+
+/**
+ * Implements hook_assemble_form_state_into_field().
+ * @param {Object} entity_type
+ * @param {String} bundle
+ * @param {String} form_state_value
+ * @param {Object} field
+ * @param {Object} instance
+ * @param {String} langcode
+ * @param {Number} delta
+ * @param {Object} field_key
+ * @return {*}
+ */
+function file_assemble_form_state_into_field(entity_type, bundle,
+                                             form_state_value,
+                                             field,
+                                             instance,
+                                             langcode,
+                                             delta,
+                                             field_key) {
+  try {
+    field_key.value = 'fid';
+    return form_state_value;
+  }
+  catch (error) {
+    console.log('file_assemble_form_state_into_field - ' + error);
+  }
+}
+
+//
+// $("#" + drupalgap_get_page_id(drupalgap_path_get())).on("pageshow", function () {
+//   document.addEventListener("deviceready", init, false);
+// });
+
+function init_media_upload() {
+  console.log("init_media_upload");
+  $("[id$=-add-media-button]").on("click", function (event) {
+    // get id of input field
+    var regExp = /(.+)-add-media-button/;
+    var input_id = regExp.exec(event.target.id)[1];
+
+    var form_id = $(this).data("form_id");
+    var name = $(this).data("name");
+    var cardinality = $(this).data("cardinality");
+    var delta = $(this).data("delta");
+
+    function setCameraOptions(srcType, medType) {
+      var options = {
+        quality: (drupalgap.settings.camera.quality) ? drupalgap.settings.camera.quality : 50,
+        sourceType: srcType, // Camera.PictureSourceType.PHOTOLIBRARY, Camera.PictureSourceType.CAMERA,
+        destinationType: Camera.DestinationType.FILE_URI,
+        mediaType: medType, // Camera.MediaType.VIDEO, Camera.MediaType.PICTURE, Camera.MediaType.ALLMEDIA
+        targetWidth: (drupalgap.settings.camera.targetWidth) ? drupalgap.settings.camera.targetWidth : 1024,
+        targetHeight: (drupalgap.settings.camera.targetHeight) ? drupalgap.settings.camera.targetHeight : 1024
+      };
+
+      return options;
+    }
+
+    function captureError(e) {
+      console.log("capture error: " + JSON.stringify(e));
+    }
+
+    function captureVideoSuccess(s) {
+      console.log("Success");
+      dpm(s);
+      console.dir(s[0]);
+      console.log("dpm:");
+      dpm(s[0]);
+      var mediaHTML = "<video  style='max-width:100%;' controls><source src='" + s[0].fullPath + "'></video>";
+      $("#" + input_id + "-media").html(mediaHTML);
+      uploadFile(s[0].fullPath);
```

Comment:
`file.toURL()` worked for me on 
[L228](https://github.com/signalpoint/DrupalGap/pull/825/commits/a2d6852a0d26ae2490a51d170646a4bb222915b0#L228)
 [L230](https://github.com/signalpoint/DrupalGap/pull/825/commits/a2d6852a0d26ae2490a51d170646a4bb222915b0#L230)
 [L239](https://github.com/signalpoint/DrupalGap/pull/825/commits/a2d6852a0d26ae2490a51d170646a4bb222915b0#L239)
 [L241](https://github.com/signalpoint/DrupalGap/pull/825/commits/a2d6852a0d26ae2490a51d170646a4bb222915b0#L241)